ui/pages/ApplicationDetail.tsx

```diff
@@ -117,6 +144,83 @@ function ApplicationDetail({ className, name }: Props) {
         app={application}
         onAuthClick={() => setGithubAuthModalOpen(true)}
       />
+      <Modal
+        //confirm modal for app removal
+        open={removeAppModalOpen}
+        onClose={() => setRemoveAppModalOpen(false)}
+        title="Are You Sure?"
+        description={`You are about to remove ${application.name} from Weave GitOps`}
+      >
+        <Flex align column center wide>
+          {authSuccess ? (
+            <Flex align>
+              <Alert severity="success" message="Authentication Successful" />
+            </Flex>
+          ) : (
+            <>
+              <Flex center>
+                <Spacer padding="small">
+                  <Alert
+                    severity="error"
+                    title="You are not Authenticated!"
+                    message="To remove this app, please authenticate with GitHub"
+                  />
+                </Spacer>
+              </Flex>
+              <Flex center>
+                <Spacer padding="small">
+                  <Button
+                    color="secondary"
+                    variant="contained"
+                    onClick={() => {
+                      setGithubAuthModalOpen(true);
+                    }}
+                  >
+                    Authenticate with GitHub
+                  </Button>
+                </Spacer>
+              </Flex>
+            </>
+          )}
+          {removeError && authSuccess && (
+            <Flex align center wide>
+              <Spacer padding="small">
+                <Alert
+                  severity="error"
+                  title="Error removing Application"
+                  message={removeError?.message}
+                />
+              </Spacer>
+            </Flex>
+          )}
+          {authSuccess && (
+            <Flex align center wide>
+              <Spacer padding="medium">
+                <Button
+                  color="secondary"
+                  variant="contained"
+                  onClick={() =>
+                    removeRequest(
+                      applicationsClient.RemoveApplication({
+                        name: application.name,
+                        namespace: application.namespace,
+                        //CAN'T FIND AUTOMERGE IN APPLICATION OBJECT
```

Comment:
Why is this comment yelling at me? :wink: 

Can you phrase this as a note for the next person to come across this code?

Reply:
I wrote that so I would remember and then I forgot about it which is an absolute classic. Will do.